src/core/network.c

```diff
@@ -489,7 +489,16 @@ int net_gethostbyaddr(IPADDR *ip, char **name)
 
 int net_ip2host(IPADDR *ip, char *host)
 {
-	return inet_ntop(ip->family, &ip->ip, host, MAX_IP_LEN) ? 0 : -1;
+	if (inet_ntop(ip->family, &ip->ip, host, MAX_IP_LEN)) {
+		return 0;
+	}
+
+	// For callers that do not check our return value and pass in an
```

Comment:
The comment is quite out of place and is better suited to the commit message.
I'd just add `host[0] = '\0'` before the `return inet_ntop` and be done with this.